packages/preview-middleware-client/src/adp/quick-actions/table-quick-action-base.ts

```diff
@@ -142,12 +147,36 @@ export abstract class TableQuickActionDefinitionBase {
         }
     }
 
+    /**
+     * Retrieves the internal table control from a UI5Element.
+     *
+     * @param table - The UI5Element instance to analyze.
+     * @returns The internal table otherwise undefined.
+     */
+    protected getInternalTable(table: UI5Element): UI5Element | undefined {
+        try {
+            let tableInternal: ManagedObject | undefined;
+
+            if (isA<SmartTable>(SMART_TABLE_TYPE, table)) {
+                const itemsAggregation = table.getAggregation('items') as ManagedObject[];
+                tableInternal = itemsAggregation.find((item) =>
+                    [M_TABLE_TYPE, TREE_TABLE_TYPE, ANALYTICAL_TABLE_TYPE, GRID_TABLE_TYPE].some((tType) =>
+                        isA(tType, item)
+                    )
+                );
+            }
+            return tableInternal as UI5Element | undefined;
+        } catch (error) {
+            return undefined;
+        }
+    }
+
     /**
      * Determines table label for the given table element
      * @param table - table element
      * @returns table label if found or 'Unnamed table'
      */
-    private getTableLabel(table: UI5Element): string {
+    public getTableLabel(table: UI5Element): string {
```

Comment:
Do we need to change this to public? I don't see it used anywhere else.

Reply:
should be private. changed.